pkg/integrations/cadvisor/cadvisor.go

```diff
@@ -0,0 +1,133 @@
+var DefaultConfig Config = Config{
+	// Common cadvisor config defaults
+	StoreContainerLabels: true,
+	ResctrlInterval:      0,
+	// Matching the default disabled set from cadvisor - https://github.com/google/cadvisor/blob/3c6e3093c5ca65c57368845ddaea2b4ca6bc0da8/cmd/cadvisor.go#L78-L93
+	DisabledMetricsSet: container.MetricSet{
+		container.MemoryNumaMetrics:              struct{}{},
+		container.NetworkTcpUsageMetrics:         struct{}{},
+		container.NetworkUdpUsageMetrics:         struct{}{},
+		container.NetworkAdvancedTcpUsageMetrics: struct{}{},
+		container.ProcessSchedulerMetrics:        struct{}{},
+		container.ProcessMetrics:                 struct{}{},
+		container.HugetlbUsageMetrics:            struct{}{},
+		container.ReferencedMemoryMetrics:        struct{}{},
+		container.CPUTopologyMetrics:             struct{}{},
+		container.ResctrlMetrics:                 struct{}{},
+		container.CPUSetMetrics:                  struct{}{},
+	},
+	EnabledMetricsSet: container.MetricSet{},
+
+	// Containerd config defaults
+	Containerd:          "/run/containerd/containerd.sock",
+	ContainerdNamespace: "k8s.io",
+}
+
+// Config controls cadvisor
+type Config struct {
+	Common config.Common `yaml:",inline"`
+
+	// Common cadvisor config options
+	// StoreContainerLabels converts container labels and environment variables into labels on prometheus metrics for each container. If false, then only metrics exported are container name, first alias, and image name.
+	StoreContainerLabels bool `yaml:"store_container_labels,omitempty"`
+
+	// WhitelistedContainerLabels list of container labels to be converted to labels on prometheus metrics for each container. store_container_labels must be set to false for this to take effect.
+	WhitelistedContainerLabels []string `yaml:"whitelisted_container_labels,omitempty"`
+
+	// EnvMetadataWhitelist list of environment variable keys matched with specified prefix that needs to be collected for containers, only support containerd and docker runtime for now.
+	EnvMetadataWhitelist []string `yaml:"env_metadata_whitelist,omitempty"`
+
+	// RawCgroupPrefixWhitelist list of cgroup path prefix that needs to be collected even when -docker_only is specified.
+	RawCgroupPrefixWhitelist []string `yaml:"raw_cgroup_prefix_whitelist,omitempty"`
+
+	// PerfEventsConfig path to a JSON file containing configuration of perf events to measure. Empty value disabled perf events measuring.
+	PerfEventsConfig string `yaml:"perf_events_config,omitempty"`
+
+	// ResctrlInterval resctrl mon groups updating interval. Zero value disables updating mon groups.
+	ResctrlInterval int `yaml:"resctrl_interval,omitempty"`
+
+	// DisableMetrics list of `metrics` to be disabled.
+	DisabledMetrics []string `yaml:"disabled_metrics,omitempty"`
+
+	// DisabledMetricsSet list of `metrics` to be disabled, in the form required by the cadvisor collector(s)
+	DisabledMetricsSet container.MetricSet
```

Comment:
is this missing a yaml tag? I'd probably make this un-exported or process it in the integration

Reply:
Not missing a tag, it was intended to be private/internal. Same with the enabled one.